daphne/src/roles_test.rs

```diff
@@ -286,3 +289,176 @@ fn hpke_decrypter() {
+    let helper = MockAggregator::new();
+    let task_id = helper.nominal_task_id();
+    let task_config = helper.get_task_config_for(task_id).unwrap();
+
+    // Construct HPKE receiver config List.
+    let hpke_receiver_config_list: Vec<HpkeReceiverConfig> =
+        serde_json::from_str(HPKE_RECEIVER_CONFIG_LIST)
+            .expect("failed to parse hpke_receiver_config_list");
+
+    // Construct HPKE config list.
+    let mut hpke_config_list = Vec::with_capacity(hpke_receiver_config_list.len());
+    for receiver_config in hpke_receiver_config_list {
+        hpke_config_list.push(receiver_config.config);
+    }
+
+    // Construct report.
+    let vdaf_config: &VdafConfig = &VdafConfig::Prio3(Prio3Config::Count);
+    let report = vdaf_config
+        .produce_report(
+            &hpke_config_list,
+            1637361337,
+            task_id,
+            DapMeasurement::U64(1),
+        )
+        .unwrap();
+
+    let req = DapRequest {
+        media_type: Some(MEDIA_TYPE_AGG_INIT_REQ),
+        payload: AggregateInitializeReq {
+            task_id: task_id.clone(),
+            agg_job_id: Id([1; 32]),
+            agg_param: b"this is an aggregation parameter".to_vec(),
+            report_shares: vec![ReportShare {
+                nonce: report.nonce.clone(),
+                extensions: report.extensions,
+                // 1st share is for Leader and the rest is for Helpers (note that there is only 1 helper).
+                encrypted_input_share: report.encrypted_input_shares[1].clone(),
+            }],
+        }
+        .get_encoded(),
+        url: task_config.helper_url.join("/aggregate").unwrap(),
+        sender_auth: Some(BearerToken::from(LEADER_BEARER_TOKEN.to_string())),
+    };
+
+    let bucket_info = BucketInfo::new(task_config, task_id, &report.nonce.clone());
+
+    // Create a new instance of report store associated to the bucket_info.
+    helper
+        .report_store
+        .lock()
+        .map_err(int_err)
+        .unwrap()
+        .deref_mut()
+        .insert(bucket_info.clone(), ReportStore::new());
+
+    // Intentionally mark report store as `collected`.
+    assert_matches!(
+        helper
+            .report_store
+            .lock()
+            .map_err(int_err)
+            .unwrap()
+            .deref_mut()
+            .get_mut(&bucket_info)
+            .unwrap()
+            .process_mark_collected(),
+        Ok(())
+    );
+
+    // Get AggregateResp and then extract the transition data from inside.
+    let agg_resp =
+        AggregateResp::get_decoded(&helper.http_post_aggregate(&req).await.unwrap().payload)
+            .unwrap();
+    let transition = &agg_resp.transitions[0];
+
+    // Expect failure due to invalid ciphertext.
+    assert_matches!(
+        transition.var,
+        TransitionVar::Failed(TransitionFailure::BatchCollected)
+    );
+}
+
+#[tokio::test]
+async fn http_post_aggregate_invalid_helper_state_store() {
```

Comment:
Thank you for the suggestion, I was also thinking the same thing.
I've refactored `roles_test.rs` so that the request generation is shared between multiple tests.